src/actions/Paste.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  Action,
+  ActionPreferences,
+  ActionReturnValue,
+  Graph,
+  TypedSelection,
+} from "../Types";
+import displayPendingEditDecorations from "../editDisplayUtils";
+import { env, Selection } from "vscode";
+import { runForEachEditor } from "../targetUtils";
+import { computeChangedOffsets } from "../computeChangedOffsets";
+import { flatten, zip } from "lodash";
+import update from "immutability-helper";
+
+export default class Paste implements Action {
+  targetPreferences: ActionPreferences[] = [{ insideOutsideType: "inside" }];
+
+  constructor(private graph: Graph) {
+    this.run = this.run.bind(this);
+  }
+
+  async run([targets]: [TypedSelection[]]): Promise<ActionReturnValue> {
+    await displayPendingEditDecorations(
+      targets,
+      this.graph.editStyles.pendingModification0,
+      this.graph.editStyles.pendingLineModification0
+    );
+
+    const text = await env.clipboard.readText();
```

Comment:
The clipboard text is also acquired in extension.ts I believe. The reason for this is that cursorless looks at clipboard contents during inference to try to automatically infer that it is a line and then automatically apply that transformation to target

The main goal was to try to mimic vim's behaviour, where if you copy a line, and then paste it somewhere, rather than trying to drop the line into the middle or the line you're on, it will put the pasted line after the current line. 

That was also the reason behind "paste" having an action position preference of "after". This way you could say "copy line air", then "paste bat" and it would paste after line bat. 

But maybe that's too much smarts; simple might be less surprising here. To approximate the behaviour, you'd say "paste after line bat" I guess.

Reply:
Yes I think it would be unexpected if the paste would be after when I wanted it to replace a token. As you say you could always add an after line if you want